File: Arrays/LeftRotateBy1/leftRotateArrayBy1.java
package Arrays.LeftRotateBy1;

public class leftRotateArrayBy1 {

    public static void rotateLeftBy1(Integer arr[]){
        Integer n = arr.length;
        Integer first = arr[0];
        for(int i = 1;i<n;i++){
            arr[i-1] = arr[i];
        }
        arr[n-1] = first;
    }

    public static void main(String args[]){
        Integer arr[] = {1,2,3,4,5};
        rotateLeftBy1(arr);
        for(Integer x: arr){
            System.out.print(x +" ");
        }
    }
}

File: Arrays/LeftRotateBy1/leftRotateArrayBy1.cpp
#include<iostream>
using namespace std;

void leftRotateBy1(int arr[], int n){
    int first = arr[0];
    for(int i = 0;i<n;i++){
        arr[i-1] = arr[i];
    }
    arr[n-1] = first;
}

int main()
{
    int arr[] = {1,2,3,4,5};
    int n = sizeof(arr)/sizeof(arr[0]);
    leftRotateBy1(arr,n);
    for(int i = 0;i< n ;i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}